content/blog/building-original-projects/index.md

```diff
@@ -0,0 +1,285 @@
+
+Your IDE will likely have a built-in debugger: learn it! For a quicker fix, you can also try print debugging using your language's print method, for example `console.log` in JavaScript.
+
+> When to use: You feel that greater understanding of your code's state at different points in its execution will help solve the problem, or after reading an error message.
+
+## SLOWLY re-read primary sources
+Sometimes what you lack is a strong enough conceptual basis to properly handle the problem. If you're having trouble applying something that you lightly skimmed or even skipped the docs for, you'll want to go back and reread it thoroughly. Even if you did read it slowly the first time, by the time you've grappled with the tech, your perspective will have changed a bit, and you might need a refresher. All docs are different, but you'll want to peek around at a few different key areas:
+
+- The table of contents. This might seem obvious, but really take a look at it. You might find a relevant article that you missed before. There may also be different sections, like a FAQ, examples, API reference, or tutorials.
+- Getting Started/Basics. True to the name, you can usually begin here. If you just wanted to jump ahead to what is relevant to you, don't forget that you can always acquire a more fundamental understand reading through the introductory articles. 
+- The API reference. Here you will find details regarding specific features, methods, objects or properties provided by your technology. If you hit a wall with the core guides, never be afraid to take a look at the information provided here.
+- The FAQs. Oftentimes, common problems will be addressed in good docs in the FAQs section. Though not all docs will have a FAQ, if they do, check it.
+- Showcase/Examples. Consists of coded implementations of the technology, this is usually where I go if I've read the main content and glanced at the API.
+
+> When to use: You feel that you're missing something, or can't understand how to use a technology for desired effect, it's likely your knowledge is has shaky foundations.
+
+## Seek the help of tech communities
+If you've entered some communities, introduce yourself and try leveraging their help in a respectful manner for new perspectives into your issue! Find the relevant channel or subforum of your community, and then communicate your problem clearly, providing relevant context, errors, and code samples. Sometimes the mere act of explaining your problem to others will grant new insight on how to solve it.
+
+Please be polite towards others when asking for help. Programming-related problems can be frustrating and at times emotional, but if people are offering to help you out of kindness, be mindful of your reaction to their advice.
```

Comment:
@AlexChadwickP wrote an awesome article for our site about this:

https://unicorn-utterances.com/posts/how-to-ask-good-questions

Maybe worth a link?